### PR TITLE
feat: add TabelCreator/TableDroper for executing create/drop table

### PR DESCRIPTION
**Pull request by ShiKaiWi**
# Which issue does this PR close?

Part of #235 

# Rationale for this change
 Refer to #235 

# What changes are included in this PR?
Add `TableCreator`/`TableDropper` to Create/Drop interpreter so that we can choose different implements of create/drop table. After this PR, two kinds of implementations will be provided: `catalog_based` and `meta_based`.

# Are there any user-facing changes?
None.

# How does this change test
Existing UT.

### CODE REVIEW

File: interpreters/src/drop/table_dropper/mod.rs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+use crate::{context::Context, drop::Result, interpreter::Output};
+
+pub mod catalog_based;
+pub mod meta_based;
+
+pub type TableDropperRef = Arc<dyn TableDropper + Send + Sync>;
+
+#[async_trait]
+pub trait TableDropper {
+    async fn drop_table(
```
Comment (Rachelint) [inline interpreters/src/drop/table_dropper/mod.rs:18]: I think `drop_table` in `TableDropper` may be a bit verbose. Same as `TableCreator`.

File: server/src/server.rs [hunk 2]
```diff
@@ -218,14 +219,25 @@ impl<Q: QueryExecutor + 'static> Builder<Q> {
...
         let function_registry = self.function_registry.context(MissingFunctionRegistry)?;
-        let instance = Instance {
-            catalog_manager,
-            query_executor,
-            table_engine,
-            function_registry,
-            limiter: self.limiter,
+
+        let instance = {
+            let creator = Arc::new(table_creator::catalog_based::TableCreatorImpl::new(
```
[the remark this answers is not in the document]
Comment (ShiKaiWi) [reply server/src/server.rs:224]: `table_creator`has been already imported as the module name.

File: interpreters/src/drop/table_dropper/mod.rs (hunk 1 above)
Comment (ShiKaiWi) [reply interpreters/src/drop/table_dropper/mod.rs:18]: It's true, and I named them as `drop/create` at first. However, the `drop` method has been used by `Drop` trait, and I have to name it `drop_table` here.

Comment (Rachelint) [reply interpreters/src/drop/table_dropper/mod.rs:18]: Why not make `Manipulator` and place drop and create logic into it?

**Review by Rachelint — APPROVED** [review]
LGTM

**Pull request merged**